flake.nix

```diff
@@ -0,0 +1,154 @@
+
+          name = cargoTOML.package.name;
+          # This is the program version.
+          version = cargoTOML.package.version;
+          # This selects a nightly Rust version, based on the date.
+          nightly-date = pkgs.lib.strings.removePrefix "nightly-" rustToolChainTOML.toolchain.channel;
+          # This is the hash of the Rust toolchain at nightly-date, required for reproducibility.
+          nightly-sha256 = "sha256-CNMj0ouNwwJ4zwgc/gAeTYyDYe0botMoaj/BkeDTy4M=";
+
+
+          # This instantiates a new Rust version based on nightly-date.
+          nightlyRustPlatform = pkgs.makeRustPlatform {
+            inherit
+              (inputs.fenix.packages.${system}.toolchainOf {
+                channel = "nightly";
+                date = nightly-date;
+                sha256 = nightly-sha256;
+              })
+              cargo rustc;
+          };
+
+          # This is a mock git program, which just returns the commit-substr value.
+          # It is called when the build process calls git. Instead of the real git,
+          # it will find this one.
+          git-mock =
+            let
+              # This evaluates to the first 7 digits of the git hash of this repo's HEAD
+              # commit, or to "dirty" if there are uncommitted changes.
+              commit-substr = builtins.substring 0 7 (inputs.self.rev or "dirty");
+            in
+            pkgs.writeShellScriptBin "git" ''
+              echo ${commit-substr}
+            '';
+
+          # srcFilter is used to keep out of the build non-source files,
+          # so that we only trigger a rebuild when necessary.
+          srcFilter = src:
+            let
+              isGitIgnored = inputs.gitignore.lib.gitignoreFilter src;
+
+              ignoreList = [
+                ".dockerignore"
+                ".envrc"
+                ".github"
+                ".travis.yml"
+                "CODE_OF_CONDUCT.md"
+                "README.md"
+                "ci"
+                "cloudbuild.yaml"
+                "codecov.yml"
+                "docker-compose.yml"
+                "rustfmt.toml"
+              ];
+            in
+            path: type:
+              isGitIgnored path type
+              && builtins.all (name: builtins.baseNameOf path != name) ignoreList;
+        in
+        rec {
+          defaultPackage = nightlyRustPlatform.buildRustPackage {
+            pname = name;
+            inherit version;
+
+            # This applies the srcFilter function to the current directory, so
+            # we don't include unnecessary files in the package.
+            src = pkgs.lib.cleanSourceWith {
+              src = ./.;
+              filter = srcFilter ./.;
+              name = "${name}-source";
+            };
+
+            # This is a hash of all the Cargo dependencies, for reproducibility.
+            cargoSha256 = "sha256-hmXhJBjc4HuyKQbxtpiIIvaL/Kl/e70sMFgdNlw4E0o=";
```

Comment:
Does this change often? Would it be worth it to add a one-liner comment with the command on how to update the hash?

Reply:
When running the nix build, it will complain and will show the new hash.